### PR TITLE
Fix Issue #980 and add test code

### PR DESCRIPTION
**Pull request by christianguenter2**
#980

### CODE REVIEW

File: src/zabapgit_unit_test.prog.abap [hunk 1]
```diff
@@ -38,6 +38,11 @@ DEFINE _append_result.
   <result>-filename = &8.
 END-OF-DEFINITION.
 
+DEFINE _given_source_is.
```
Comment (larshp) [inline src/zabapgit_unit_test.prog.abap]: can we move this macro to a local method instead? The code will not look as pretty, but in this case it is easy to get rid of the macro, also chaining should mainly be used for declarations, https://help.sap.com/http.svc/rc/abapdocu_751_index_htm/7.51/en-US/abenchained_statements_guidl.htm

**Comment by christianguenter2** [comment]
I changed the macro to a method. Now we get this build error.
image

This is line 2609
https://github.com/larshp/abapGit/blob/1c7e4c9fe055635af5a24defc11a57fb6d569e7c/src/zabapgit_unit_test.prog.abap#L2609

Not quite sure what the issue is.

**Comment by larshp** [comment]
I'll take a look tomorrow

**Comment by larshp** [comment]
opened https://github.com/larshp/abaplint/issues/187

**Pull request merged**